pkgs/development/compilers/edk2/default.nix

```diff
@@ -25,8 +43,10 @@ edk2 = stdenv.mkDerivation {
 
   buildInputs = [ libuuid pythonEnv ];
 
-  makeFlags = [ "-C BaseTools" ];
-  NIX_CFLAGS_COMPILE = "-Wno-return-type -Wno-error=stringop-truncation";
+  makeFlags = [ "-C BaseTools" ]
+    ++ lib.optional (stdenv.isDarwin) [ "BUILD_CC=clang BUILD_CXX=clang++ BUILD_AS=clang" ];
+
+  NIX_CFLAGS_COMPILE = "-Wno-return-type" + lib.optionalString (!stdenv.isDarwin) " -Wno-error=stringop-truncation";
```

Comment:
```suggestion
  NIX_CFLAGS_COMPILE = "-Wno-return-type" + lib.optionalString (!stdenv.cc.isClang) " -Wno-error=stringop-truncation";
```

Reply:
Actually, I took the liberty to go with `stdenv.cc.isGNU` then, because it makes more sense to me. Let me know if I should go back to the negated one.

Reply:
Right.